pkg/pipelines/config/validate.go

```diff
@@ -46,6 +48,18 @@ func (vv *validateVisitor) validateServiceURLs() []error {
 }
 
 func (vv *validateVisitor) Environment(env *Environment) error {
+	if vv.manifest.Config != nil {
+		if vv.manifest.Config.CICD != nil {
+			if vv.manifest.Config.CICD.Namespace == env.Name {
+				vv.errs = append(vv.errs, errors.New("Cannot add a new application to Config environments"))
+			}
+		}
+		if vv.manifest.Config.Argo != nil {
+			if vv.manifest.Config.Argo.Namespace == env.Name {
+				vv.errs = append(vv.errs, errors.New("Cannot add a new application to Config environments"))
+			}
+		}
+	}
```

Comment:
This check you have here is checking for namespace collision between any plain old Environment and "specail" CICD/ArgoCD which is good validation.   However, we got here not because we are adding any applications so the error messages are  off.    Also, we should use the error functions like other errors which logs the env.Path (otherwise, good luck troubleshooting what went wrong with the manifest file.)

Instead of keeping manifest in vw, I would keep the namespaces of "config" (i.e. CICD and ArgoCD namespaces).  Then, your checking code here wouldn't need to reference the namespaces from manifest.   It will make the code looks clener.   It means that you should walk the Config in config.walk() which is another problem.   You need to validate the names of CICD and ArgoCD environments which have been skipped now.